compiler/z/codegen/OMRTreeEvaluator.cpp

```diff
@@ -14117,6 +14117,168 @@ bool isSettingOp(uint8_t byteValue, TR::InstOpCode::Mnemonic SI_opcode)
    }
 
 
+
+TR::Register *
+OMR::Z::TreeEvaluator::bitpermuteEvaluator(TR::Node *node, TR::CodeGenerator *cg)
+   {
+
+   TR_ASSERT(node->getNumChildren() == 3, "Wrong number of children in bitpermuteEvaluator");
+   TR::Node *value = node->getChild(0);
+   TR::Node *addr = node->getChild(1);
+   TR::Node *length = node->getChild(2);
+
+   bool nodeIs64Bit = node->getSize() == 8;
+   auto valueReg = cg->evaluate(value); // this will either be a constant or a load
+   auto addrReg = cg->evaluate(addr);   // this will either be a constant or a load
+
+   TR::Register *tmpReg = cg->allocateRegister(TR_GPR);
+
+   //Zero result reg
+   TR::Register *resultReg = cg->allocateRegister(TR_GPR);
+   generateRRInstruction(cg, TR::InstOpCode::getXORRegOpCode(), node, resultReg, resultReg);
+
+   if (length->getOpCode().isLoadConst())
+      {
+      // Manage the constant length case
+      uintptrj_t arrayLen = length->getIntegerNodeValue<uintptrj_t>();
+      for (uintptrj_t x = 0; x < arrayLen; ++x)
+         {
+         // zero tmpReg
+         generateRRInstruction(cg, TR::InstOpCode::getXORRegOpCode(), node, tmpReg, tmpReg);
+
+         //load the shift amount into tmpReg
+         //TODO: Check to see if we need sourceMR->stopUsingMemRefRegister(cg)
```

Comment:
Technically yes, though in practice we have code that handles this automatically.

Reply:
Ah okay. Then I'll remove these from here.